static/js/components/ObservationPlanRequestForm.jsx

```diff
@@ -0,0 +1,288 @@
+    allocationList.length === 0 ||
+    !selectedAllocationId ||
+    Object.keys(instrumentFormParams).length === 0
+  ) {
+    return <h3>No robotic instruments available...</h3>;
+  }
+
+  if (
+    !allGroups ||
+    allGroups.length === 0 ||
+    telescopeList.length === 0 ||
+    instrumentList.length === 0
+  ) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  if (
+    !allGroups ||
+    allGroups.length === 0 ||
+    telescopeList.length === 0 ||
+    instrumentList.length === 0
+  ) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  const groupLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  allGroups?.forEach((group) => {
+    groupLookUp[group.id] = group;
+  });
+
+  const telLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  telescopeList?.forEach((tel) => {
+    telLookUp[tel.id] = tel;
+  });
+
+  const allocationLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  allocationList?.forEach((allocation) => {
+    allocationLookUp[allocation.id] = allocation;
+  });
+
+  const instLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  instrumentList?.forEach((instrumentObj) => {
+    instLookUp[instrumentObj.id] = instrumentObj;
+  });
+
+  const handleSelectedAllocationChange = (e) => {
+    setSelectedAllocationId(e.target.value);
+  };
+
+  const handleSelectedLocalizationChange = (e) => {
+    setSelectedLocalizationId(e.target.value);
+  };
+
+  const handleSubmit = async ({ formData }) => {
+    setIsSubmitting(true);
+    const json = {
+      gcnevent_id: gcnevent.id,
+      allocation_id: selectedAllocationId,
+      localization_id: selectedLocalizationId,
+      target_group_ids: selectedGroupIds,
+      payload: formData,
+    };
+    await dispatch(gcnEventActions.submitObservationPlanRequest(json));
+    setIsSubmitting(false);
+  };
+
+  const validate = (formData, errors) => {
+    if (
+      formData.start_date &&
+      formData.end_date &&
+      formData.start_date > formData.end_date
+    ) {
+      errors.start_date.addError("Start Date must come before End Date");
+    }
+
+    return errors;
+  };
+
+  return (
+    <div className={classes.container}>
+      <InputLabel id="allocationSelectLabel">Allocation</InputLabel>
+      <Select
+        inputProps={{ MenuProps: { disableScrollLock: true } }}
+        labelId="allocationSelectLabel"
+        value={selectedAllocationId}
+        onChange={handleSelectedAllocationChange}
+        name="followupRequestAllocationSelect"
+        className={classes.allocationSelect}
+      >
+        {allocationList?.map((allocation) => (
+          <MenuItem
+            value={allocation.id}
+            key={allocation.id}
+            className={classes.allocationSelectItem}
+          >
+            {`${
+              telLookUp[instLookUp[allocation.instrument_id].telescope_id].name
```

Comment:
should be `telLookUp` without `instLookUp`?

Reply:
Since the allocation is on an instrument, I think we want the instrument named.

Reply:
OK I see that now.